### ISSUE TITLE
 Backwards incompatible in multiple port service

### ISSUE THREAD

**Issue by kylinsoong** [F5Networks/k8s-bigip-ctlr#2255]
At the end of #1464,  Backwards incompatible issue has be raised, I would like to create another issue for tracking this issue.

The following are copied from #1340 

**Backwards incompatible issue** - In Previous version, the servicePort is optional, but in later version servicePort is mandatory, which cause upgrade issue, we spent logs of time add hub mode back for smooth upgrade, but now blocked on servicePort.

Could you help to re-investigate, at least made the servicePort is optional? made the velidation logic like：if the servicePort match the targetPort then use the new added logic, or else, keep the previous logic?

**Comment by trinaths** [comment]
PR submitted to added to internal JIRA board for team to review.
